master/internal/db/postgres_trial_metrics.go

```diff
@@ -106,13 +106,32 @@ func BunSelectMetricsQuery(metricGroup model.MetricGroup, inclArchived bool) *bu
 // BunSelectMetricGroupNames sets up a bun select query for getting all the metric group and names.
 func BunSelectMetricGroupNames() *bun.SelectQuery {
 	return Bun().NewSelect().Table("trials").
-		ColumnExpr("DISTINCT jsonb_object_keys(summary_metrics) as json_path").
+		ColumnExpr("jsonb_object_keys(summary_metrics) as json_path").
 		ColumnExpr("jsonb_object_keys(summary_metrics->jsonb_object_keys(summary_metrics))" +
 			" as metric_name").
 		Where("summary_metrics IS NOT NULL").
 		Order("json_path").Order("metric_name")
 }
 
+// BunUpdateMetricMean calculates summary metrics mean, and return as.
+// id | summary_metrics
+//
+//	1 | { avg_metrics: { accuracy: { min: 1, max: 2, last: 2, mean: 1 } } }
+func BunUpdateMetricMean() *bun.SelectQuery {
+	return Bun().NewSelect().Table("trials").
+		TableExpr("jsonb_object_keys(summary_metrics) as metric_group").
+		Column("trials.id").
+		ColumnExpr(`jsonb_object_agg(
+	metric_group,
+		(SELECT jsonb_object_agg(m.key, CASE WHEN m.value -> 'type' = '"number"'::jsonb
+			THEN m.value - '{sum, count}'::text[] ||
+				jsonb_build_object(
+					'mean', (m.value ->> 'sum')::float8 / (m.value ->> 'count')::int)
```

Comment:
pretty sure mean is also not overflow safe

Reply:
By overflow, do you mean the result can be infinity? or the decimals can overflow?